tests/ctest/test_opmphm_vheap.c

```diff
@@ -0,0 +1,278 @@
+	Vheap * h = elektraVheapInit (mincomp, 4);
+	exit_if_fail (h, "vheap init error");
+	succeed_if (!elektraVheapRemove (NULL), "remove empty working");
+	elektraVheapDel (h);
+
+	succeed_if (!elektraVheapInsert (NULL, NULL), "insert NULL working");
+}
+
+
+static void test_empty ()
+{
+	Vheap * h = elektraVheapInit (mincomp, 4);
+	exit_if_fail (h, "vheap init error");
+	succeed_if (elektraVheapIsEmpty (h), "should be empty");
+	elektraVheapInsert (h, NULL);
+	succeed_if (!elektraVheapIsEmpty (h), "should not be empty");
+	elektraVheapRemove (h);
+	succeed_if (elektraVheapIsEmpty (h), "should be empty");
+	elektraVheapDel (h);
+}
+
+static void test_data_max_ordered ()
+{
+	Vheap * h = elektraVheapInit (maxcomp, 100);
+	exit_if_fail (h, "vheap init error");
+	int data[100];
+	for (int i = 0; i < 99; ++i)
+	{
+		data[i] = i;
+		succeed_if (elektraVheapInsert (h, &data[i]), "insert error");
+	}
+	int max = 100;
+	for (int i = 0; i < 99; ++i)
+	{
+
+		void * dv = elektraVheapRemove (h);
+		succeed_if (dv, "remove error");
+		int d = *(int *)dv;
+		succeed_if (d < max, "ascending error");
+		max = d;
+	}
+	elektraVheapDel (h);
+
+	h = elektraVheapInit (maxcomp, 100);
+	exit_if_fail (h, "vheap init error");
+	for (int i = 99; i >= 0; --i)
+	{
+		data[i] = i;
+		succeed_if (elektraVheapInsert (h, &data[i]), "insert error");
+	}
+	max = 100;
+	for (int i = 0; i < 99; ++i)
+	{
+		void * dv = elektraVheapRemove (h);
+		succeed_if (dv, "remove error");
+		int d = *(int *)dv;
+		succeed_if (d < max, "descending error");
+		max = d;
+	}
+	elektraVheapDel (h);
+}
+
+static void test_data_min_ordered ()
+{
+	Vheap * h = elektraVheapInit (mincomp, 100);
+	exit_if_fail (h, "vheap init error");
+	int data[100];
+	for (int i = 0; i < 99; ++i)
+	{
+		data[i] = i;
+		succeed_if (elektraVheapInsert (h, &data[i]), "insert error");
+	}
+	int min = -1;
+	for (int i = 0; i < 99; ++i)
+	{
+		void * dv = elektraVheapRemove (h);
+		succeed_if (dv, "remove error");
+		int d = *(int *)dv;
+		succeed_if (d > min, "ascending error");
+		min = d;
+	}
+	elektraVheapDel (h);
+
+	h = elektraVheapInit (mincomp, 100);
+	exit_if_fail (h, "vheap init error");
+	for (int i = 99; i >= 0; --i)
+	{
+		data[i] = i;
+		succeed_if (elektraVheapInsert (h, &data[i]), "insert error");
+	}
+	min = -1;
+	for (int i = 0; i < 99; ++i)
+	{
+		void * dv = elektraVheapRemove (h);
+		succeed_if (dv, "remove error");
+		int d = *(int *)dv;
+		succeed_if (d > min, "descending error");
+		min = d;
+	}
+	elektraVheapDel (h);
+}
+
+static void test_grow_shrink ()
+{
+	int maxElem = 101;
+	int data = 42;
+	for (int minSize = 1; minSize <= 100; ++minSize)
+	{
+		int actualSize = minSize;
+		Vheap * h = elektraVheapInit (mincomp, minSize);
+		exit_if_fail (h, "vheap init error");
+		for (int i = 1; i <= maxElem; ++i)
+		{
+			succeed_if (elektraVheapInsert (h, &data), " insert error");
+			if (i > actualSize)
+			{
+				// grow
+				actualSize <<= 1;
```

Comment:
I think you can just use `*=2` here. There is no need to hand optimize a test, as far as I know.